commit e0c81b5ba2f59b0a28378aadd2c3ed47ea38c025 (refs/changes/59/1859/3)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2017-04-27T16:15:34+00:00 (2 years, 5 months ago)
    
    DOCKER-1044 Specifying triton.network.public label doesn't imply an external NIC
    Reviewed by: Todd Whiteman <todd.whiteman@joyent.com>
    Approved by: Todd Whiteman <todd.whiteman@joyent.com>

diff --git a/docs/api/features/networks.md b/docs/api/features/networks.md
index 08e5bec..0317cb8 100644
--- a/docs/api/features/networks.md
+++ b/docs/api/features/networks.md
@@ -53,6 +53,10 @@ The external network used by a container can be changed by setting the
 docker run --label triton.network.public=external2
 ```
 
+Note that this this only overrides the default public network selection. This
+means that when fabric networks are enabled you will still need to specify one
+of `-p` or `-P` to get the public NIC.
+
 
 ## Related
 
diff --git a/test/integration/api-create.test.js b/test/integration/api-create.test.js
index 80154c9..ca507f1 100644
--- a/test/integration/api-create.test.js
+++ b/test/integration/api-create.test.js
@@ -868,18 +868,17 @@ test('run external network (docker run --label triton.network.public=)',
         }, oncreate);
 
         function oncreate(err, result) {
-            var extNic;
             var nics = result.vm.nics;
             if (FABRICS) {
                 // Expect two nics, one fabric and one external.
-                t.equal(nics.length, 2, 'two nics');
-                extNic = (nics[0].nic_tag === 'external' ? nics[0] : nics[1]);
+                t.equal(nics.length, 1, 'only one nic');
+                t.equal(nics[0].nic_tag.indexOf('sdc_overlay'), 0,
+                    'no external network, only fabric');
             } else {
                 t.equal(nics.length, 1, 'only one nic');
-                extNic = nics[0];
+                t.equal(nics[0].network_uuid, externalNetwork.uuid,
+                    'correct external network');
             }
-            t.equal(extNic.network_uuid, externalNetwork.uuid,
-                'correct external network');
             DOCKER_ALICE.del('/containers/' + result.id + '?force=1', ondelete);
         }
 
